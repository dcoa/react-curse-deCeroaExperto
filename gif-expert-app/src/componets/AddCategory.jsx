import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({ setCategory }) => {
    const [inputValue, setinputValue] = useState('');

    const handleSubmit = (e) => {
        // eslint-disable-next-line no-unused-expressions
        e.preventDefault();
        if( inputValue.trim().length > 2){
            setCategory(categ => [inputValue, ...categ])
            setinputValue('')
        }
    }
    return (
        <>
            <form onSubmit={handleSubmit}>
                <input
                    type="text"
                    value={inputValue}
                    placeholder='Escribe lo que quieres buscar y presiona "Enter"'
                    onChange={e => setinputValue(e.target.value)}
                />
            </form>
        </>
    )
}

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}
