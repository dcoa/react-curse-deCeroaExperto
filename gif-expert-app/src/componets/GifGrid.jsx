import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifts } from '../hooks/useFetchGifts';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ Category }) => {
    // const [images, setImages] = useState([]);
    const { data, loading } = useFetchGifts(Category);
    //transformaci√≥n para crear una galeria de imagenes
    let columns = [];

    for (let i = 0; i < data.length; i += 3) {
        columns.push(data.slice(i, i + 3))
    }

    return (
        <>
            <h3>{Category}</h3>
            <div className='card-grid'>
                {loading && 'Cargandoo...'}
                {
                    columns.map((images, i) => <div key={'columns' + i} className='columns'>
                        {images.map(img => <GifGridItem key={img.id} {...img} />)}
                    </div>)
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    Category: PropTypes.string.isRequired
}
