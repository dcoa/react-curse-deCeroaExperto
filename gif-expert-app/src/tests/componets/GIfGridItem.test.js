import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from './../../componets/GifGridItem'

describe('Test Gif grid item component', () => {
    
    const title = 'Hisoka'
    const url = 'https://media2.giphy.com/media/IurNlhZHaj7TG/giphy.gif?cid=0d1d774aq1fu4tgx2iu14xoqgouduy4l29blwubi4j2icdha&rid=giphy.gif&ct=g'
    const wrapper = shallow(<GifGridItem title={title} url={url}/>)

    test('should render correct', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should render title in a <p>', () => {
         const p = wrapper.find('p');
         expect(p.text().trim()).toBe(title);    
    })
    
    test('should render a <img> with src and alt attributes', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('should the container has class card', () => {
        const div = wrapper.find('div');
        expect(div.hasClass('card')).toBe(true);

    })
});
