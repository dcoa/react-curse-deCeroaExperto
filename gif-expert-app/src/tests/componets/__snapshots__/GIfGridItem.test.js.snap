// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Gif grid item component should render correct 1`] = `
<div
  className="card"
>
  <img
    alt="Hisoka"
    src="https://media2.giphy.com/media/IurNlhZHaj7TG/giphy.gif?cid=0d1d774aq1fu4tgx2iu14xoqgouduy4l29blwubi4j2icdha&rid=giphy.gif&ct=g"
  />
  <p>
    Hisoka
  </p>
</div>
`;
